vignettes/rcbc.Rmd

```diff
@@ -0,0 +1,251 @@
+---
+title: "rcbc: CBC bindings for R"
+output:
+  rmarkdown::html_vignette:
+    toc: false
+    fig_caption: true
+    self_contained: yes
+fontsize: 11pt
+documentclass: article
+vignette: >
+  %\VignetteIndexEntry{rcbc: CBC bindings for R}
+  %\VignetteEngine{knitr::rmarkdown_notangle}
+---
+
+```{r, include = FALSE}
+# skip running this vignette during R package checks
+# this is useful to reduce package check times on CRAN
+is_check <- ("CheckExEnv" %in% search()) || any(c("_R_CHECK_TIMINGS_",
+             "_R_CHECK_LICENSE_") %in% names(Sys.getenv()))
+knitr::opts_chunk$set(fig.align = "center", eval = !is_check)
+
+# create default status variable for R package checks (used later)
+status <- "optimal"
+```
+
+```{r, include = FALSE}
+# load package for generating vignette
+devtools::load_all()
```

Comment:
I've included `devtools::load()` here so that the package vignette is rebuilt using the developmental/latest version of the package. Although one could just install the developmental version of the package locally before trying to rebuild the vignettes, I've found this approach easier to work with. But that's just my opinion - I'm happy to remove this if you want?

Also, technically, devtools isn't detected as a dependency because the vignette has some R code that will stop all code chunks from being evaluated if the vignette code is being run/evaluated as part of R package checks (see https://github.com/jeffreyhanson/rcbc/blob/vign/vignettes/rcbc.Rmd#L15-L23). This is mainly beneficial if (1) the time needed to run the code in the vignette exceeds the limit imposed by CRAN (see https://stackoverflow.com/questions/28961431/computationally-heavy-r-vignettes), or (2) you want to include code in vignette that can't run on CRAN (e.g. let's say you wanted to include some code in a vignette that required installation of an R package not available on CRAN, such as gurobi). I don't think (2) is relevant here, but I just wanted to note that for completeness.

I can easily remove this extra stuff if you don't think it's worth it though?

Reply:
Interesting. Does it really take that long to run the vignette? I thought the vignette would always be built with the current package version when it is built (CRAN, pkgdown etc.). But I rarely write vignettes :)

My gut feeling says to not do that unless really needed.

Reply:
Yeah that's a good point. I rebuilt the vignette locally and recorded that it took only 40 seconds to rebuild. So, it looks like all this extra stuff to handle computationally heavy vignettes isn't needed in this particular case. I'll remove the `devtools::load_all()` and extra stuff so that the vignette follows standard practices.

Reply:
Ok, I've just removed all the extra stuff so that the vignette follows standard practices. I've also tweaked the .gitignore and documentation workflow to ensure that junk files won't be committed to the repo (i.e `./README.html`, `.github/depends.rds` and `.github/R-version`.